elements/includes/creative_commons.inc

```diff
@@ -77,287 +130,1022 @@ function xml_form_elements_creative_commons($element, FormStateInterface $form_s
+    $element['license_fieldset']['license_type']['#required'] = ($values['license_fieldset']['license_type'] != 'manual');
+    $element['license_fieldset']['license_uri']['#required'] = ($values['license_fieldset']['license_type'] == 'manual');
+  }
+
+  // Replace the description of the type chooser when none was chosen.
+  if ($element['license_fieldset']['license_type']['#value'] == 'none') {
+    $element['license_fieldset']['license_type']['#description'] = t('Note that your selection will still output a blank element in the resulting XML, so make sure to use a cleanup template to remove it.');
+  }
+}
+
 /**
  * Ajax callback to render the CC license.
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ *
+ * @return array
+ *   The markup to replace the target (wrapper) with.
  */
-function xml_form_elements_creative_commons_ajax(array $form, FormStateInterface $form_state) {
+function xml_form_elements_creative_commons_ajax(array &$form, FormStateInterface $form_state) {
   $parents = $form_state->getTriggeringElement()['#parents'];
   array_pop($parents);
   $creative_commons_element = NestedArray::getValue($form, $parents);
 
-  return $creative_commons_element['license_output'];
+  return $creative_commons_element;
 }
 
 /**
- * Ask the creative commons REST API for a license.
+ * Submit handler for graceful no-JS degradation.
  *
- * Left as a function for use by community modules.
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function xml_form_elements_creative_commons_license_update_submit(array $form, FormStateInterface $form_state) {
+  // If we have to manually update the form, force rebuild.
+  $form_state->setRebuild();
+}
+
+/**
+ * Validator for the `creative_commons` element.
  *
- * @param string $commercial
- *   String toggle 'y' to allow 'n' to disallow.
+ * @param array $element
+ *   The form element.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ * @param array $form
+ *   The form.
+ */
+function xml_form_elements_creative_commons_validate(array &$element, FormStateInterface $form_state, array $form) {
+  // Make sure creative commons form errors get displayed.
+  $validation_sections =& drupal_static('form_set_error' . ':limit_validation_errors');
+  $original_validation_sections = $validation_sections;
+  $validation_sections = [[$element['#name']]];
+  // Populate element.
+  _xml_form_elements_creative_commons_populate_element($element, $form_state);
+  // If a license is required, error if none was chosen.
+  if (($element['license_fieldset']['license_type']['#value'] == 'none') && $element['license_fieldset']['license_type']['#required']) {
+    form_set_error($element['#name'] . '][license_fieldset][license_type', t('@element is required.', ['@element' => $element['license_fieldset']['license_type']['#title']]));
```

Comment:
[`form_set_error()` isn't a thing in D8](https://api.drupal.org/api/drupal/includes%21form.inc/function/form_set_error/7.x)... should be replaced with [`FormState::setError()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21FormState.php/function/FormState%3A%3AsetError/8.7.x) or [`FormState::setErrorByName()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21FormState.php/function/FormState%3A%3AsetErrorByName/8.7.x)?

... Appears to be more occurrences than just that here... will just comment here, though.